li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -325,82 +313,101 @@ public void unsubscribe() {
 
   @Override
   public void commitSync() {
-    throw new UnsupportedOperationException("Not implemented yet");
+    commitSync(Duration.ofMillis(_defaultApiTimeoutMs));
   }
 
   @Override
   public void commitSync(Duration timeout) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    executeCallbackWithTimeout("commitSync", null,
+        (consumer, ignored, timeoutDuration) -> consumer.commitSync(timeoutDuration), timeout);
   }
 
   @Override
   public void commitSync(Map<TopicPartition, OffsetAndMetadata> offsets) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    commitSync(offsets, Duration.ofMillis(_defaultApiTimeoutMs));
   }
 
   @Override
   public void commitSync(Map<TopicPartition, OffsetAndMetadata> offsets, Duration timeout) {
-    throw new UnsupportedOperationException("Not implemented yet");
+   executeCallbackWithTimeout("commitSync", getPartitionValueMapByCluster(offsets),
+        (consumer, perClusterOffsets, timeoutDuration) -> {
+      consumer.commitSync((Map<TopicPartition, OffsetAndMetadata>) perClusterOffsets, timeoutDuration);
+        }, timeout);
   }
 
   @Override
   public void commitAsync() {
-    throw new UnsupportedOperationException("Not implemented yet");
+    commitAsync(null);
   }
 
   @Override
   public void commitAsync(OffsetCommitCallback callback) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    for (ClusterConsumerPair<K, V> entry : getImmutableConsumerList()) {
+      entry.getConsumer().commitAsync(callback);
+    }
   }
 
   @Override
   public void commitAsync(Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCommitCallback callback) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    List<ClusterConsumerPair<K, V>> consumers = getImmutableConsumerList();
+    for (Map.Entry<ClusterDescriptor, Map<TopicPartition, OffsetAndMetadata>> entry :
+        getPartitionValueMapByCluster(offsets).entrySet()) {
+      getOrCreatePerClusterConsumer(consumers, entry.getKey()).commitAsync(entry.getValue(), callback);
+    }
   }
 
   @Override
   public void seek(TopicPartition partition, long offset) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    getOrCreateConsumerForTopic(partition.topic()).seek(partition, offset);
   }
 
   @Override
   public void seekToBeginning(Collection<TopicPartition> partitions) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    List<ClusterConsumerPair<K, V>> consumers = getImmutableConsumerList();
+    for (Map.Entry<ClusterDescriptor, Collection<TopicPartition>> entry : getPartitionsByCluster(partitions).entrySet()) {
+      getOrCreatePerClusterConsumer(consumers, entry.getKey()).seekToBeginning(entry.getValue());
```

Comment:
not sure if seek() should result in the creation of a consumer - i think only subscribe/assign should

Reply:
The current implementation of config reload closes clients but does not recreate them. They will be recreated at the subsequent operation. If we keep this "lazy" reload behavior, seek() may need to create a consumer.

Reply:
i dont know what the lazy approach gets us - we expect these clients to be active.